src/components/Map/Map.tsx

```diff
@@ -121,59 +123,103 @@ export default function Map(props: MapProps): JSX.Element {
           properties,
           sourceId,
         });
-        updateFeatureState(selectStateId, 'select', { sourceId, id });
+        updateFeatureState(selectStateId, 'select', [{ sourceId, id }]);
       }
     },
     [updateFeatureState, selectStateId]
   );
 
   const drawFocusTo = useCallback(
-    (mapEntity: MapEntityId) => {
-      const map: any = mapRef?.current;
-      if (!map || !mapEntity.id || !mapEntity.sourceId) {
+    (mapEntity: MapEntityId[]) => {
+      if (mapEntity.length <= 0) {
         return;
       }
-      const foundSource = geoData && geoData.find((geo) => geo.id === mapEntity.sourceId);
-      if (!foundSource || !foundSource.data) {
+      const map: any = mapRef?.current;
+      if (!map) {
         return;
       }
-      const feature = foundSource.data.features.find((featureData: any) => featureData.id === mapEntity.id);
-      if (!feature) {
-        return;
+      const coordinates: MapGeometry = [];
+      mapEntity.forEach((entity) => {
+        if (!entity.id || !entity.sourceId) {
+          return;
+        }
+        const foundSource = geoData && geoData.find((geo) => geo.id === entity.sourceId);
+        if (!foundSource || !foundSource.data) {
+          return;
+        }
+        const feature = foundSource.data.features.find((featureData: any) => featureData.id === entity.id);
+        if (!feature) {
+          return;
+        }
+        coordinates.push(feature.geometry.coordinates);
+      });
+      if (coordinates.length > 0) {
+        const bbox = MapService.getBoundingBox([coordinates]);
+        map.fitBounds([bbox.lowerLeft, bbox.upperRight], { padding: 20 });
       }
-      const coordinates = feature.geometry.coordinates;
-      const bbox = MapService.getBoundingBox([[coordinates]]);
-      map.fitBounds([bbox.lowerLeft, bbox.upperRight], { padding: 20 });
     },
     [geoData]
   );
 
-  const onLoad = () => {
+  useEffect(() => {
```

Comment:
This useEffect subscribes and, in its cleanup, unsubscribes event handlers. This allows us to update the fill colors for hover states when the interactive layer changes (this was previously handled only in the onLoad function)

Reply:
Nice 👍